solr-template/src/main/java/eu/xenit/ass/fqwarmup/solrtemplate/WarmupListenerXmlTransformer.java

```diff
@@ -0,0 +1,121 @@
+package eu.xenit.ass.fqwarmup.solrtemplate;
+
+import eu.xenit.ass.trial.fqwarmup.FilterQueryWarmupListener;
+import eu.xenit.ass.trial.fqwarmup.SortWarmupListener;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+public class WarmupListenerXmlTransformer {
+
+    public void transForm(InputStream in, OutputStream out) {
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder;
+
+        try {
+            dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(in);
+            doc.getDocumentElement().normalize();
+
+            Element config = doc.getDocumentElement();
+            Element query = (Element) config.getElementsByTagName("query").item(0);
+
+            addFilterQueryWarmupListener(doc, query, "newSearcher");
+            addFilterQueryWarmupListener(doc, query, "firstSearcher");
+
+            addSortWarmupListener(doc, query, "newSearcher");
+            addSortWarmupListener(doc, query, "firstSearcher");
+
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(doc);
+            StreamResult result = new StreamResult(out);
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.transform(source, result);
+
+        } catch (ParserConfigurationException e) {
```

Comment:
These exceptions should probably be thrown upwards so it exits with a nonzero exitcode instead of failing silently